addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
Martin,

the issue is that non of the values are integers. LEDStateDTO inherits HSBType and uses for hue, saturation and brightness an BigDecimal.
I agree that using float for calculation wasn't the best idea... Double's would have been better but for sure not first get an integer while the original was a BigDecimal and then do floating point math....

FoHere is an example with the original code the below will output:

> 		LEDStateDTO ledState = LEDStateDTO.valueOf(0, 0, 0, 128, 4, 200, 0, 0);
> 		System.out.println(new Color(128,4,200).toString());
> 		System.out.println(ledState.getColor().toString());
> 

> java.awt.Color[r=128,g=4,b=200]
> java.awt.Color[r=124,g=4,b=199]

Perhaps the value differences are minimal, but when you start fading you will see little jumps because of these rounding errors. This get's very noticeable because LEDStateDTO will get re-loaded from the device before each call and for each 'brightness' change, you get different RGB values..

An alternative for this function could be:

>     public Color getColor() {
>         return new Color(super.getRGB());
>     }

Note:super.getRGB() will return a int representing a RGB value that newColor is more than happy to accept.

A other alternative is to remove the getColor method altogether and get the RGB using bit shifting. Usually I won't change a public API But if you are ok with it so I will be.... As far as I can see in the build is that LEDStateDTO is not exported in the OSGi bundle...

Before:
```
            byte r = (byte) (ledState.getColor().getRed() & 0xFF);
            byte g = (byte) (ledState.getColor().getGreen() & 0xFF);
            byte b = (byte) (ledState.getColor().getBlue() & 0xFF);
```

after:
```
            byte r = (byte) (ledState.getRGB()>>16 & 0xFF);
            byte g = (byte) (ledState.getRGB()>>8 & 0xFF);
            byte b = (byte) (ledState.getRGB() & 0xFF);
```
For Reference this is the complete function:
https://github.com/rvt/openhab2-addons/blob/master/addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/ClassicWiFiLEDDriver.java#L140

Hope I made myself more clear...